presto-geospatial/src/test/java/com/facebook/presto/geospatial/TestGeoQueries.java

```diff
@@ -0,0 +1,451 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.geospatial;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.testing.LocalQueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import com.facebook.presto.tpch.TpchConnectorFactory;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.metadata.FunctionExtractor.extractFunctions;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static com.facebook.presto.tpch.TpchMetadata.TINY_SCHEMA_NAME;
+
+public class TestGeoQueries
+        extends AbstractTestQueryFramework
+{
+    public TestGeoQueries()
+    {
+        super(TestGeoQueries::createLocalQueryRunner);
+    }
+
+    @Test
+    public void testSTPoint()
+            throws Exception
+    {
+        assertQuery("select ST_AsText(ST_Point(1, 4))", "select 'POINT (1 4)'");
+        assertQuery("select ST_AsText(ST_Point(122.3, 10.55))", "select 'POINT (122.3 10.55)'");
+    }
+
+    @Test
+    public void testSTLineFromText()
+            throws Exception
+    {
+        assertQuery("select ST_AsText(ST_LineFromText('LINESTRING EMPTY'))", "select 'MULTILINESTRING EMPTY'");
```

Comment:
It looks like ST_AsText(ST_Point(input)) doesn't preserve the type of the input geometry when geometry is empty. Here LineString got converted to MultiLineString. Further down in this file I also see that Polygon gets converted to MultiPolygon. Is this expected? Perhaps, worth documenting.

Reply:
yes, this is expected. Since when ST_AsText trying to deserialize, it only knows it is a polyline, not clear whether it is a line or multiline. I will document it.